### PR TITLE
ENG-3651 Prevent the creation of a user with guest username

### PR DESCRIPTION
**Pull request by alepintus**

### CODE REVIEW

File: admin-console/src/main/java/org/entando/entando/apsadmin/user/UserAction.java [hunk 1]
```diff
@@ -71,6 +71,9 @@ protected void checkAddUserUser() {
 		String username = this.getUsername();
 		String profileTypeCode = this.getProfileTypeCode();
 		try {
+			if (username.equalsIgnoreCase("guest")) {
```
Comment (zonia3000) [inline admin-console/src/main/java/org/entando/entando/apsadmin/user/UserAction.java]: You can use `SystemConstants.GUEST_USER_NAME`

File: engine/src/main/java/org/entando/entando/web/user/validator/UserValidator.java [hunk 2]
```diff
@@ -154,6 +156,10 @@ public void validateUserPost(UserRequest request, BindingResult bindingResult) {
         if (!request.getPassword().equals(request.getPasswordConfirm())) {
             bindingResult.reject(UserValidator.ERRCODE_PASSWORD_CONFIRM_MISMATCH, new String[]{}, "user.password-confirm.mismatch");
         }
+        if (username.equalsIgnoreCase("guest")) {
```
Comment (zonia3000) [inline engine/src/main/java/org/entando/entando/web/user/validator/UserValidator.java]: `SystemConstants.GUEST_USER_NAME`

Comment (alepintus) [reply engine/src/main/java/org/entando/entando/web/user/validator/UserValidator.java]: done

File: admin-console/src/main/java/org/entando/entando/apsadmin/user/UserAction.java (hunk 1 above)
Comment (alepintus) [reply admin-console/src/main/java/org/entando/entando/apsadmin/user/UserAction.java]: done

File: admin-console/src/main/java/org/entando/entando/apsadmin/user/package_en.properties [hunk 3]
```diff
@@ -58,6 +58,7 @@ note.userConfirm.trash.goBack=If you don''t want to delete this user, go back to
 note.user.intro.html=<p>The access to information, using services and performing tasks are regulated by permissions, roles and groups associated to each User.</p><p>A User represents an entity that has the ability to perform tasks and access to information.</p><p>This capacity is determined by all roles and groups.</p><p>This section provides the features to create, edit or delete users, access to the list of existing users, view their status and manage permissions.</p>
 
 error.user.duplicateUser=User ''{0}'' already exists
+error.user.invalidGuestUsername=Guest is a reserved username inside Entando
```
Comment (cecchisandrone) [inline admin-console/src/main/java/org/entando/entando/apsadmin/user/package_en.properties]: guest

Comment (alepintus) [reply admin-console/src/main/java/org/entando/entando/apsadmin/user/package_en.properties]: done

**Review by zonia3000 — APPROVED** [review]

**Review by cecchisandrone — APPROVED** [review]

**Pull request merged**